src/lavinmq/queue/message_store.cr

```diff
@@ -359,7 +359,7 @@ module LavinMQ
         @segments.reject! do |seg, mfile|
           next if seg == current_seg # don't the delete the segment still being written to
 
-          if @segment_msg_count[seg] == @acks[seg].size // sizeof(UInt32)
+          if (acks = @acks[seg]?) && @segment_msg_count[seg] == (acks.size // sizeof(UInt32))
```

Comment:
Just tried to make it more readable, don't know if i succeeded... 

```suggestion
          if @acks.has_key?(seg) && @segment_msg_count[seg] == @acks[seg].size // sizeof(UInt32)
```

```suggestion
          if @segment_msg_count[seg] == (@acks[seg]?.try(&.size) || 0) // sizeof(UInt32)
```

Reply:
I think I did the way I did to get rid of one access to the instance variable. Imo the `has_key?` check is the most readable, but maybe
```crystal
if @segment_msg_count[seg] == (@acks.fetch(seg, 0) // sizeof(UInt32))
```
is as good. `try` and `||` isn't very readable imo.

Reply:
Yes, even better, thanks!